src/components/auth/PrivateRoute.tsx

```diff
@@ -0,0 +1,17 @@
+import React from 'react';
+import {useSelector} from 'react-redux';
+import {Route, Redirect, RouteProps} from 'react-router-dom';
+
+import {RootState} from '../../store';
+
+interface Props extends RouteProps {
+    component: any;
+}
+
+export const PrivateRoute = ({component: Component, ...rest}:Props) => {
```

Comment:
Space after the `:` - `{component: Component, ...rest}: Props`.

If you want to use the children way:

```
export const PrivateRoute = ({component, children}: React.PropsWithChildren) => {}
``` 

Also, don't type annotate the expanded fields. Just `{component}: Props` is sufficient.

Reply:
Turns out not, because the type of `component` will be used for rendering.